src/ui/server/controllers/insurance/all-sections/index.test.ts

```diff
@@ -0,0 +1,82 @@
+import { get } from '.';
+import { PAGES } from '../../../content-strings';
+import { ROUTES, TEMPLATES } from '../../../constants';
+import insuranceCorePageVariables from '../../../helpers/page-variables/core/insurance';
+import { mockReq, mockRes } from '../../../test-mocks';
+import { Request, Response } from '../../../../types';
+import api from '../../../api';
+
+describe('controllers/insurance/all-sections', () => {
+  let req: Request;
+  let res: Response;
+
+  const mockReferenceNumber = '10001';
+
+  const mockApplication = {
+    referenceNumber: mockReferenceNumber,
+  };
+
+  const mockGetApplicationResponse = mockApplication;
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+
+    req.params.referenceNumber = mockReferenceNumber;
+  });
+
+  describe('get', () => {
+    let getApplicationSpy = jest.fn(() => Promise.resolve(mockGetApplicationResponse));
+
+    beforeEach(() => {
+      api.keystone.getApplication = getApplicationSpy;
+    });
+
+    it('should call api.keystone.getApplication', async () => {
+      await get(req, res);
+
+      expect(getApplicationSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should render template', async () => {
+      await get(req, res);
+
+      const expectedVariables = {
+        ...insuranceCorePageVariables({
+          PAGE_CONTENT_STRINGS: PAGES.INSURANCE.START,
+          BACK_LINK: req.headers.referer,
+        }),
+        application: mockApplication,
+      };
+
+      expect(res.render).toHaveBeenCalledWith(TEMPLATES.INSURANCE.ALL_SECTIONS, expectedVariables);
```

Comment:
minor but could we dereference templates on the top and make them super neat, then we can also only have to change in one place if the template name changes?

Reply:
Great idea, have added `TEMPLATE` to this controller and we can use this approach for all controllers moving forward? Will create a ticket to update previously created controllers to use this approach